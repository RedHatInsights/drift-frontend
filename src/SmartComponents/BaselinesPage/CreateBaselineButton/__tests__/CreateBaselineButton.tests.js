import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedCreateBaselineButton, { CreateBaselineButton } from '../CreateBaselineButton';

describe('CreateBaselineButton', () => {
    let props;

    beforeEach(() => {
        props = {
            addSystemModalOpened: false,
            loading: false,
            permissions: {
                baselinesWrite: true
            }
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <CreateBaselineButton { ...props }/>
        );

        expect(wrapper.find('[id="create-baseline-button"]').prop('isDisabled')).toBe(false);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render disabled with no write permissions', () => {
        props.permissions.baselinesWrite = false;
        const wrapper = shallow(
            <CreateBaselineButton { ...props }/>
        );

        expect(wrapper.find('[id="create-baseline-button"]').prop('isDisabled')).toBe(true);
        expect(wrapper.find('Tooltip')).toHaveLength(1);
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <CreateBaselineButton { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        it('should call toggleCreateBaselineModal', () => {
            const toggleCreateBaselineModal = jest.fn();
            const toggleAddSystemModal = jest.fn();
            const history = { location: { pathname: '/baselines' }, push: jest.fn() };
            const wrapper = mount(
                <CreateBaselineButton
                    history={ history }
                    toggleCreateBaselineModal={ toggleCreateBaselineModal }
                    toggleAddSystemModal={ toggleAddSystemModal }
                    { ...props }
                />
            );

            wrapper.find('.pf-c-button').simulate('click');
            expect(toggleCreateBaselineModal).toHaveBeenCalledTimes(1);
            expect(toggleAddSystemModal).toHaveBeenCalledTimes(0);
        });

        it('should call toggleAddSystemModal', () => {
            props.addSystemModalOpened = true;
            const toggleCreateBaselineModal = jest.fn();
            const toggleAddSystemModal = jest.fn();
            const history = { location: { pathname: '/' }, push: jest.fn() };
            const wrapper = mount(
                <CreateBaselineButton
                    history={ history }
                    toggleCreateBaselineModal={ toggleCreateBaselineModal }
                    toggleAddSystemModal={ toggleAddSystemModal }
                    { ...props }
                />
            );

            wrapper.find('.pf-c-button').simulate('click');
            expect(toggleCreateBaselineModal).toHaveBeenCalledTimes(1);
            expect(toggleAddSystemModal).toHaveBeenCalledTimes(1);
        });
    });
});

describe('ConnectedCreateBaselineButton', () => {
    let initialState;
    let mockStore;
    let props;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            addSystemModalState: {
                addSystemModalOpened: false
            }
        };

        props = {
            permissions: {
                baselinesWrite: true
            }
        };
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedCreateBaselineButton { ...props } />
                </Provider>
            </MemoryRouter>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should dispatch toggleCreateBaselineModal', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedCreateBaselineButton { ...props } />
                </Provider>
            </MemoryRouter>
        );

        const actions = store.getActions();
        wrapper.find('.pf-c-button').simulate('click');
        expect(actions).toEqual([{ type: 'TOGGLE_CREATE_BASELINE_MODAL' }]);
    });

    it('should dispatch toggleAddSystemModal', () => {
        initialState.addSystemModalState.addSystemModalOpened = true;
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedCreateBaselineButton { ...props } />
                </Provider>
            </MemoryRouter>
        );

        const actions = store.getActions();
        wrapper.find('.pf-c-button').simulate('click');
        expect(actions).toEqual([
            { type: 'OPEN_ADD_SYSTEM_MODAL' },
            { type: 'TOGGLE_CREATE_BASELINE_MODAL' }
        ]);
    });
});
