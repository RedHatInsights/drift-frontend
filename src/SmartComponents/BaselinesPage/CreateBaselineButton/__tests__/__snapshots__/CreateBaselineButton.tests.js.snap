// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedCreateBaselineButton should render correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(CreateBaselineButton))
        permissions={
          Object {
            "baselinesWrite": true,
          }
        }
      >
        <Connect(CreateBaselineButton)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          permissions={
            Object {
              "baselinesWrite": true,
            }
          }
        >
          <CreateBaselineButton
            addSystemModalOpened={false}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            permissions={
              Object {
                "baselinesWrite": true,
              }
            }
            toggleAddSystemModal={[Function]}
            toggleCreateBaselineModal={[Function]}
          >
            <Button
              id="create-baseline-button"
              onClick={[Function]}
              ouiaId="create-baseline-button"
              variant="primary"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                data-ouia-component-id="create-baseline-button"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                id="create-baseline-button"
                onClick={[Function]}
                role={null}
                type="button"
              >
                Create baseline
              </button>
            </Button>
          </CreateBaselineButton>
        </Connect(CreateBaselineButton)>
      </withRouter(Connect(CreateBaselineButton))>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`CreateBaselineButton should render correctly 1`] = `
<Fragment>
  <Button
    id="create-baseline-button"
    isDisabled={false}
    onClick={[Function]}
    ouiaId="create-baseline-button"
    variant="primary"
  >
    Create baseline
  </Button>
</Fragment>
`;

exports[`CreateBaselineButton should render mount correctly 1`] = `
<CreateBaselineButton
  addSystemModalOpened={false}
  loading={false}
  permissions={
    Object {
      "baselinesWrite": true,
    }
  }
>
  <Button
    id="create-baseline-button"
    isDisabled={false}
    onClick={[Function]}
    ouiaId="create-baseline-button"
    variant="primary"
  >
    <button
      aria-disabled={false}
      aria-label={null}
      className="pf-c-button pf-m-primary"
      data-ouia-component-id="create-baseline-button"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="create-baseline-button"
      onClick={[Function]}
      role={null}
      type="button"
    >
      Create baseline
    </button>
  </Button>
</CreateBaselineButton>
`;
